### PR TITLE
add patchStrategy for lists in SPDX document

### PR DESCRIPTION
**Pull request by matthyx**
should I run something to regenerate some code?

### CODE REVIEW

**Comment by vladklokun** [comment]
Yup, whenever we touch the types, we have to re-generate the code. The procedure is described [here](https://github.com/kubescape/storage#changes-to-the-types).

File: pkg/apis/softwarecomposition/v1beta1/spdx_types.go [hunk 1]
```diff
@@ -557,11 +557,17 @@ type Document struct {
...
+	// +patchStrategy=merge
+	Packages []*Package `json:"packages"  patchStrategy:"merge" patchMergeKey:"name"`
+	// +patchMergeKey=fileName
+	// +patchStrategy=merge
+	Files         []*File         `json:"files" patchStrategy:"merge" patchMergeKey:"fileName"`
 	OtherLicenses []*OtherLicense `json:"hasExtractedLicensingInfos"`
-	Relationships []*Relationship `json:"relationships"`
+	// +patchMergeKey=spdxElementId
+	// +patchStrategy=merge
+	Relationships []*Relationship `json:"relationships"  patchStrategy:"merge" patchMergeKey:"spdxElementId"`
```
[the remark this answers is not in the document]
Comment (matthyx) [reply pkg/apis/softwarecomposition/v1beta1/spdx_types.go:570]: yes

File: vendor/modules.txt [hunk 2]
```diff
@@ -56,6 +56,9 @@ github.com/go-openapi/jsonreference/internal
 # github.com/go-openapi/swag v0.22.3
 ## explicit; go 1.18
 github.com/go-openapi/swag
+# github.com/go-test/deep v1.1.0
+## explicit; go 1.16
+github.com/go-test/deep
```
Comment (vladklokun) [inline vendor/modules.txt:61]: issue: we use `go-cmp` here already, we can use it or `testify/assert`instead of pulling another dependency

**Review by vladklokun — CHANGES_REQUESTED** [review]

**Comment by Bezbran** [comment]
@avrahams Do you have a solution to "how to test API locally / unitary"?

**Pull request merged**